cmd/arrai/build.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/arr-ai/arrai/pkg/arrai"
+	"github.com/arr-ai/arrai/pkg/arraictx"
+	"github.com/arr-ai/arrai/pkg/ctxfs"
+	"github.com/arr-ai/arrai/rel"
+	"github.com/arr-ai/arrai/syntax"
+	"github.com/arr-ai/arrai/tools"
+	"github.com/arr-ai/wbnf/parser"
+	"github.com/spf13/afero"
+	"github.com/urfave/cli/v2"
+)
+
+var compileCommand = &cli.Command{
+	Name:    "compile",
+	Aliases: []string{"compile"},
+	Usage:   "compile arrai scripts into a runnable binary",
+	Action:  compile,
+	Flags: []cli.Flag{
+		outFlag,
+	},
+}
+
+func compile(c *cli.Context) error {
+	tools.SetArgs(c)
+	file := c.Args().Get(0)
+
+	return compileFile(arraictx.InitRunCtx(context.Background()), file, c.Value("out").(string))
+}
+
+func compileFile(ctx context.Context, path, out string) error {
+	if err := runFileExists(ctx, path); err != nil {
+		return err
+	}
+
+	bundledScripts := bytes.Buffer{}
+	if err := bundleFiles(ctx, path, &bundledScripts, ""); err != nil {
+		return err
+	}
+
+	if out == "" {
+		out = strings.TrimSuffix(filepath.Base(path), filepath.Ext(path))
+	}
+
+	f, err := ctxfs.SourceFsFrom(ctx).Create(out)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	if err := buildBinary(ctx, bundledScripts.Bytes(), f); err != nil {
+		return err
+	}
+	return nil
+}
+
+func createGoFile(ctx context.Context, bundledScripts []byte) ([]byte, error) {
+	template, err := Asset("internal/build/main.arrai")
```

Comment:
Nit: Should this really be using arr.ai? It looks like a very simple Go template would do the job with much less fuss?

Reply:
I just thought of using arrai and I didn't really know how to use go template 😅

Reply:
Give it a shot I reckon; it would save us from all of the go-bindata cruft, and probably even be less source code as well. Not sure how to encode the bytes, but possible something like `NewNumber(b).Repr()` is an option.